### PR TITLE
Added docker file/action

### PR DESCRIPTION
**Pull request by AhmedKorim**

### CODE REVIEW

File: .github/workflows/dockerise.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+name: Docker Build
+
+on:
+  push:
+    # Publish `master` as Docker `latest` image.
+    branches:
+      - master
```
Comment (shekohex) [inline .github/workflows/dockerise.yml:7]: we need also branch `develop` under an `edge` image tag.

File: .github/workflows/dockerise.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            ~/.cargo/git
+            ~/.cargo/bin
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Run Install Chef
+        uses: actions-rs/cargo@v1
+        with:
+          command: install
+          args: cargo-chef
```
Comment (shekohex) [inline .github/workflows/dockerise.yml:45]: we don't have to use `cargo-chef` here, we would better use `cross` from cargo

**Review by shekohex — CHANGES_REQUESTED** [review]

File: dockerfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+FROM rust as runtime
+WORKDIR app
+VOLUME /files
+VOLUME /images
+
+COPY --from=builder /app/target/release/node-template /usr/local/bin
+RUN /usr/local/bin/node-template purge-chain --dev -y
+
+ENTRYPOINT ["/usr/local/bin/node-template"]
```
Comment (shekohex) [inline dockerfile:27]: I guess we would mostly not going to use these stuff, we could build the `node-template` on the host (Github action) and ship it with the image itself.

File: .github/workflows/dockerise.yml (hunk 2 above)
Comment (drewstone) [reply .github/workflows/dockerise.yml:45]: so multiple docker files?

File: .github/workflows/dockerise.yml (hunk 1 above)
Comment (drewstone) [reply .github/workflows/dockerise.yml:7]: so multiple dockerfiles? We should ensure we can keep branches separate right?

Comment (shekohex) [reply .github/workflows/dockerise.yml:7]: no, in the same docker file.

Comment (shekohex) [reply .github/workflows/dockerise.yml:7]: Example:
```yml
branches:
    - master
    - develop
```

and later


```
[ "$VERSION" == "master" ] && VERSION=latest
[ "$VERSION" == "develop" ] && VERSION=edge
```

**Comment by shekohex** [comment]
Closing this PR in favor of #131

**Pull request closed without merging**